torchtext/datasets/text_classification.py

```diff
@@ -0,0 +1,410 @@
+
+    raw_folder = os.path.join(raw_folder, dataset_name.lower() + '_csv')
+
+    if os.path.exists(processed_folder) is not True:
+        os.makedirs(processed_folder)
+
+    src_filepath = os.path.join(raw_folder, 'train.csv')
+    tgt_filepath = os.path.join(processed_folder, dataset_name + '.train')
+    text_normalize(src_filepath, tgt_filepath)
+
+    src_filepath = os.path.join(raw_folder, 'test.csv')
+    tgt_filepath = os.path.join(processed_folder, dataset_name + '.test')
+    text_normalize(src_filepath, tgt_filepath)
+
+    print("Dataset %s preprocessed." % dataset_name)
+
+
+def load_text_classification_data(filepath, fields, ngrams=1):
+    """Load train/test data from a file and generate data
+        examples with ngrams.
+    """
+
+    def label_text_processor(line, fields, ngrams=1):
+        """Process text string and generate examples for dataset."""
+        fields = [('text', fields['text']), ('label', fields['label'])]
+        label, text = line.split(",", 1)
+        label = float(label.split("__label__")[1])
+        ex = data.Example.fromlist([text, label], fields)
+        tokens = ex.text[1:]  # Skip the first space '\t'
+        ex.text = generate_ngrams(tokens, ngrams)
+        return ex
+
+    examples = []
+    with open(filepath) as src_data:
+        for line in tqdm(src_data):
+            examples.append(label_text_processor(line, fields, ngrams))
+    return examples
+
+
+def iters(train_examples, test_examples, fields, sort_key,
+          split_ratio=0.7, batch_size=32, device='cpu', random_state=None):
+    """Create iterator objects for splits of the dataset.
+
+    Arguments:
+        split_ratio: split train_examples into train set (split_ratio)
+            and valid set (1-split_ratio). Default: 0.7
+        batch_size: batch size. Default: 32
+        device: the device to sent data. Default: 'cpu'
+        random_state: the random state provided by user. Default: None
+
+    Examples:
+        >>> train_iter, test_iter, valid_iter = txt_cls.iters(device="cpu")
+
+    Outputs:
+        - train_iter: a iterator based on
+            train.csv file with split_ratio percent.
+        - train_iter: a iterator based on test.csv file.
+        - valid_iter: a iterator based on
+            train.csv file with 1-split_ratio percent.
+
+    """
+
+    rnd = dataset.RandomShuffler(random_state)
+    train_ratio, valid_ratio = split_ratio, 1 - split_ratio
+    train_examples, _test_examples, valid_examples = \
+        dataset.rationed_split(train_examples, train_ratio, 0.0,
+                               valid_ratio, rnd)
+    train = TextDataset(train_examples, fields)
+    train.sort_key = sort_key
+    test = TextDataset(test_examples, fields)
+    test.sort_key = sort_key
+    valid = TextDataset(valid_examples, fields)
+    valid.sort_key = sort_key
+
+    return generate_iterators(
+        (train, test, valid), [batch_size] * 3, device=device)
+
+
+class TextDataset(torch.utils.data.Dataset):
```

Comment:
Do we still need this?

Reply:
I actually include it here as a new abstract class. Later on, I can simply merge it into TextClassification Dataset.